### ISSUE TITLE
Kotlin script caching does not work with Gradle Enterprise remote build cache

### ISSUE THREAD

**Issue by ldaley** [gradle/kotlin-dsl#1023]
### Expected Behavior

To be able to use our cache.

### Current Behavior

Fails with:

```
Could not load entry build file '«script»' from remote build cache: Loading entry from 'https://«host»/cache/gradle-kotlin-dsl/dx7phw87epyvb7zje5dti3tvl' response status 400: Bad Request
```

The problem is that `gradle-kotlin-dsl/dx7phw87epyvb7zje5dti3tvl` is rejected as a cache key. The GE build cache expects keys to be `[a-z0-9]{1,128}`. We could lift this restriction, but I think it would be better to change the key format that Kotlin is using here.

### Steps to Reproduce (for bugs)

Just try to use it on gradle/gradle, with access to the Gradle Inc. cache instance enabled.

**Comment by mark-vieira** [comment]
FWIW, relaxing this restriction would likely be significantly complicated in a multi-node replication scenario running different versions of controller/nodes.

+1 on just aligning the cache key produced.

**Comment by bamboo** [comment]
Fixed in 438c2e1d795fa424d2c67ef3116cc920a294ab8c by discarding the `gradle-kotlin-dsl/` cache key prefix.
